internal/cloud/backend_plan.go

```diff
@@ -95,7 +87,25 @@ func (b *Cloud) opPlan(stopCtx, cancelCtx context.Context, op *backend.Operation
 		return nil, diags.Err()
 	}
 
-	return b.plan(stopCtx, cancelCtx, op, w)
+	// If the run errored, exit before checking whether to save a plan file
+	run, err := b.plan(stopCtx, cancelCtx, op, w)
+	if err != nil {
+		return run, err
+	}
+
+	// Maybe save plan file
+	if op.PlanOutPath != "" {
+		bookmark := cloudplan.NewSavedPlanBookmark(run.ID, b.hostname)
+		err = bookmark.Save(op.PlanOutPath)
+	}
+
+	// Only display next steps if everything succeeded
+	if err == nil {
+		// Cloud currently supports plan -out but not genconfig
+		op.View.PlanNextStep(op.PlanOutPath, "")
+	}
+
+	return run, err
```

Comment:
Same suggestion here. I think we want to be clear about returning nil if we are also returning an error.